packages/jaeger-ui/src/components/Monitor/ServicesView/serviceGraph.tsx

```diff
@@ -87,30 +79,20 @@ export class ServiceGraphImpl extends React.PureComponent<TProps> {
 
       this.getData().forEach((line: ServiceMetricsObject, idx: number) => {
         graphs.push(
-          <AreaSeries
+          <Area
             key={i++}
-            data={line.metricPoints ? line.metricPoints : []}
-            getNull={(d: Points) => d.y !== null}
-            onNearestX={(_datapoint: Points, { index }: { index: number }) => {
-              this.setState({
-                crosshairValues: this.getData().map((d: ServiceMetricsObject) => ({
-                  ...d.metricPoints[index],
-                  label: d.quantile,
-                })),
-              });
-            }}
-            opacity={0.1}
-            color={[color || this.colors[idx]]}
+            type="monotone"
+            dataKey="y"
+            connectNulls
+            fillOpacity={0.1}
+            stroke={color || this.colors[idx]}
+            fill={color || this.colors[idx]}
           />
         );
         graphs.push(
-          <LineSeries
-            getNull={(d: Points) => d.y !== null}
-            key={i++}
-            data={line.metricPoints ? line.metricPoints : []}
-            color={[color || this.colors[idx]]}
-          />
+          <Line connectNulls key={i++} type="monotone" dataKey="y" stroke={color || this.colors[idx]} />
         );
+        data.push(line.metricPoints ? line.metricPoints : []);
```

Comment:
That isn't the right way of doing this. 
How would this data be supplied to the components without a rerender?

Reply:
That's a valid point. I'm replacing it with state.